### PR TITLE
x-pack/filebeat/module/microsoft: drop empty api responses

### LINKED ISSUE #34786: Filebeat module for microsoft needs to drop empty api responses
This issue was also affecting the integration in versions prior to 8.6 and the issue is detailed here:

https://github.com/elastic/integrations/issues/5148

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

Allows dropping of responses that hold no event data.

This is a port of elastic/integrations#5164.

## Why is it important?

Empty responses waste space and do not conform to the expected format of the datastream.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Fixes #34786

### CODE REVIEW

File: x-pack/filebeat/module/microsoft/defender_atp/config/atp.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@ request.transforms:
 
 response.split:
   target: body.value
+  ignore_empty_value: true
```
Comment (andrewkroh) [inline x-pack/filebeat/module/microsoft/defender_atp/config/atp.yml:32]: Do we have any test cases in our logs that have an empty `value` field?

**Review by andrewkroh — APPROVED** [review]
LGTM. If there's not a sample log test case it would be nice to have one (but actually now that I think about this, I have no idea if the test harness would support verifying drops).

File: x-pack/filebeat/module/microsoft/defender_atp/config/atp.yml (hunk 1 above)
Comment (efd6) [reply x-pack/filebeat/module/microsoft/defender_atp/config/atp.yml:32]: We do in the integration, but not here.

Comment (efd6) [reply x-pack/filebeat/module/microsoft/defender_atp/config/atp.yml:32]: I've come up with an approach that serves to test this. It's not pretty, but it work.

File: x-pack/filebeat/module/microsoft/defender_atp/test/defender_atp-test.json.log [hunk 2]
```diff
@@ -2,3 +2,4 @@
...
+{"value":[],"note":"THIS MESSAGE SHOULD NOT END UP IN THE EXPECTS JSON FILE."}
```
Comment (efd6) [inline x-pack/filebeat/module/microsoft/defender_atp/test/defender_atp-test.json.log:5]: If this ends up in the result we error out. This is the case if the pipeline drop processors are not in place.

Comment (CyberTaoFlow) [reply x-pack/filebeat/module/microsoft/defender_atp/test/defender_atp-test.json.log:5]: if you would like here is a sample of an empty event for your test case
```


{"@timestamp":"2023-03-31T23:06:13.507Z","@metadata":{"beat":"filebeat","type":"_doc","version":"8.4.3","pipeline":"filebeat-8.4.3-microsoft-defender_atp-pipeline"},"agent":{"type":"filebeat
","version":"8.4.3","ephemeral_id":"382fa7a9-285e-47b9-bedc-badbd225347e","id":"e20495e9-3358-4155-b4c9-9a519775d234","name":"collector-01"},"fileset":{"name":"defender_atp"},"service":{"typ
e":"microsoft"},"ecs":{"version":"1.12.0"},"input":{"type":"httpjson"},"organization":{"name":"acme"},"json":{"@odata.context":"https://api.securitycenter.windows.com/api/$metadata#Alerts","
value":[]},"event":{"created":"2023-03-31T23:06:13.507Z","dataset":"microsoft.defender_atp","module":"microsoft"},"message":"{\"@odata.context\":\"https://api.securitycenter.windows.com/api/
$metadata#Alerts\",\"value\":[]}","tags":["acme","defender-atp","forwarded"]}

**Pull request merged**